### ISSUE TITLE
Add support for MatchesPrefix and MatchesSuffix in StorageBucket resource

### ISSUE THREAD

**Issue by kaspersjo** [GoogleCloudPlatform/k8s-config-connector#690]
### Checklist

- [X] I did not find a related open enhancement request.
- [X] I understand that enhancement requests filed in the GitHub repository are by default low priority.
- [X] If this request is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Describe the feature or resource

Add support for setting `MatchesPrefix` and `MatchesSuffix` lifecycle conditions on storage buckets.

### Additional information

https://cloud.google.com/storage/docs/lifecycle#matchesprefix-suffix

**Comment by jcanseco** [comment]
Thanks. This is one of our Terraform-based resources, and it looks like there is already a request for Terraform to support this feature: https://github.com/hashicorp/terraform-provider-google/issues/11990

Once Terraform implements support, we should be able to bring that into KCC when we do our regular upgrade of our Terraform dependency.

**Comment by gcaracuel** [comment]
https://github.com/hashicorp/terraform-provider-google-beta/pull/4527 has been merged so I think this task is unblocked :)

**Comment by jcanseco** [comment]
Awesome :)

We should be able to bring this in on our next upgrade of our Terraform dependency. We'll keep you posted.

**Comment by tokatsu-gc** [comment]
+1 My customer would like to use this feature, too.
